addons/crm/tests/test_crm_lead.py

```diff
@@ -365,3 +366,70 @@ def test_mailgateway(self):
+        self.assertEqual(leads[0:6], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '3010')])
+        self.assertEqual(leads[0:6], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '010')])
+        self.assertEqual(leads[0:6], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '+32485112233')])
+        self.assertEqual(leads[10] + leads[11], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '485112233')])
+        self.assertEqual(leads[10] + leads[11], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '+320')])
+        self.assertEqual(leads[12], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '+3200')])
+        self.assertEqual(leads[12], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '(089) / 636-48018')])
+        self.assertEqual(leads[7], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '    08963648018')])
+        self.assertEqual(leads[7], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '08963648018    ')])
+        self.assertEqual(leads[7], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '089+6364+8018')])
+        self.assertEqual(leads[7], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '0 89 6 364 80 18')])
+        self.assertEqual(leads[7], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '(089)/63(hello world)6-48018')])
+        self.assertEqual(leads[7], searched_leads - existing_leads)
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '0-8-9-6-3-6')])
+        self.assertEqual(leads[7], searched_leads - existing_leads)
+
+        # Errors:
+
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '')])
+        self.assertFalse(bool(searched_leads))
+        searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '+')])
+        self.assertFalse(bool(searched_leads))
+
+        for term in ['9', '42', '+1', '+13', '+00']:
+            try:
+                searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', term)])
+                self.assertTrue(False)
+            except UserError:
+                pass
```

Comment:
Since we only add a simple check, let's make this test a lot shorter.

```py
    def test_phone_filter_input(self):
        leads = self.env['crm.lead'].create([{
            'name': 'Lead 1',
            'phone': '+32499112233',
        }, {
            'name': 'Lead 2',
            'phone': '0032499112233',
        }])

        with self.assertRaises(UserError):
            # search term containing only characters and no digits should throw an error
            self.env['crm.lead'].search([('phone_mobile_search', 'like', 'coucou')])

        with self.assertRaises(UserError):
            # search term containing less than 3 digits should throw an error
            self.env['crm.lead'].search([('phone_mobile_search', 'like', '+32')])

        self.assertEqual(
            leads,
            self.env['crm.lead'].search([('phone_mobile_search', 'like', '499112233')])
        )
```